htdocs/fourn/class/fournisseur.product.class.php

```diff
@@ -491,47 +491,48 @@ function find_min_price_product_fournisseur($prodid, $qty=0)
-        $sql.= $this->db->plimit(1);
+        $product_fourn_list = $this->list_product_fournisseur_price($prodid);
 
         dol_syslog(get_class($this)."::find_min_price_product_fournisseur", LOG_DEBUG);
 
-        $resql = $this->db->query($sql);
-        if ($resql)
+        if (is_array($product_fourn_list))
         {
-            $record = $this->db->fetch_array($resql);
-            $this->product_fourn_price_id	= $record["product_fourn_price_id"];
-            $this->fourn_ref				= $record["ref_fourn"];
-            $this->fourn_price				= $record["price"];
-            $this->fourn_qty				= $record["quantity"];
-            $this->fourn_remise_percent		= $record["remise_percent"];
-            $this->fourn_remise				= $record["remise"];
-            $this->fourn_unitprice			= $record["unitprice"];
-            $this->fourn_charges			= $record["charges"];
-            $this->fourn_unitcharges		= $record["unitcharges"];
-            $this->fourn_tva_tx				= $record["tva_tx"];
-            $this->fourn_id					= $record["fourn_id"];
-            $this->fourn_name				= $record["supplier_name"];
-            $this->fk_price_expression      = $record["fk_price_expression"];
-            $this->id						= $prodid;
-            $this->db->free($resql);
-            return 1;
+            if (count($product_fourn_list) == 0) 
+            {
+                return 0;
+            }
+            else
+            {
+                $min = -1;
+                foreach($product_fourn_list as $productfourn)
+                { 
+                    if ($productfourn->fourn_price < $min || $min == -1)
+                    {
+                        $this->product_fourn_price_id   = $productfourn->product_fourn_price_id;
+                        $this->fourn_ref                = $productfourn->fourn_ref;
+                        $this->fourn_price              = $productfourn->fourn_price;
+                        $this->fourn_qty                = $productfourn->fourn_qty;
+                        $this->fourn_remise_percent     = $productfourn->fourn_remise_percent;
+                        $this->fourn_remise             = $productfourn->fourn_remise;
+                        $this->fourn_unitprice          = $productfourn->fourn_unitprice;
+                        $this->fourn_charges            = $productfourn->fourn_charges;
+                        $this->fourn_unitcharges        = $productfourn->fourn_unitcharges;
+                        $this->fourn_tva_tx             = $productfourn->fourn_tva_tx;
+                        $this->fourn_id                 = $productfourn->fourn_id;
+                        $this->fourn_name               = $productfourn->fourn_name;
+                        $this->fk_supplier_price_expression      = $productfourn->fk_supplier_price_expression;
+                        $this->id                       = $productfourn->id;
+                    }
+                }
+                return 1;
+            }
```

Comment:
Why not add priceparcer to 'find_min_price_product_fournisseur();'
Because parced price could not be the minimum anymore?